[ADD] l10n_co_currency_rate_update: Currency Rate from Superfinanciera

yanchirino opened the pull request:
- Add automatic USD/COP exchange rate update from Superfinanciera
- Integrate with Superfinanciera SOAP web service
- Create cron job to update rates every 12 hours
- Support multi-company rate updates
- Add detailed logging for rate operations
- Include HTML description for Odoo Apps Store

dreispt commented on l10n_co_trm/models/__init__.py:
```diff
@@ -0,0 +1 @@
+from . import res_currency  # noqa
```
Why the noqa?
```suggestion
from . import res_currency
```

yanchirino replied:
# noqa was used to silence the “unused import” warning, but it has been removed now — thanks for the suggestion.

dreispt commented on l10n_co_trm/models/res_currency.py:
```diff
@@ -0,0 +1,77 @@
+                    ]
+                )
+
+                if obj_rate.name is False:
```
Make it pythonic
```suggestion
                if not obj_rate:
```

yanchirino replied:
done.

dreispt commented on l10n_co_trm/models/res_currency.py:
```diff
@@ -0,0 +1,77 @@
+    def get_trm(self):
+        rate = self.get_rate()
+        # rate_amount = 1 / rate
+        date_now = time.strftime("%Y-%m-%d")
```
Suggestion: rename to `date_today`? `now`makes me think is is a datetime.
```suggestion
        date_today = time.strftime("%Y-%m-%d")
```

yanchirino replied:
done.

dreispt commented on l10n_co_currency_rate_update/__manifest__.py:
```diff
@@ -0,0 +1,14 @@
+    "auto_install": False,
+    "application": False,
+    "installable": True,
+}
```
You are missing the `readme` fragments with at least a DESCRIPTION.rst and USAGE.rst.
Those will generate the OCA README.

yanchirino replied:
done.

dreispt commented on l10n_co_trm/models/res_currency.py:
```diff
@@ -0,0 +1,77 @@
+        # rate_amount = 1 / rate
+        date_now = time.strftime("%Y-%m-%d")
+        companies = self.env["res.company"].search([])
+        try:
```
This `try` does not seem useful, as it does not really handle the exception.
Can it be removed?

yanchirino replied:
done.

dreispt reviewed (commented):
Some minor improvement suggestions.

dreispt commented on l10n_co_trm/static/description/index.html:
```diff
@@ -0,0 +1,462 @@
+<li>Integración con el servicio web SOAP de la Superintendencia Financiera de Colombia</li>
+<li>Creación automática de registros de tasa de cambio para todas las empresas</li>
+<li>Actualización de tasas existentes cuando ya existe un registro para la fecha</li>
+<li>Logging detallado de las operaciones de actualización</li>
```
This will be automatically generated by the precommit hook, based on the readme fragments.

yanchirino replied:
done.

dreispt commented:
Thank you.
The README header dos not translate properly to HTML.
I believe the issue is the `\` leading character in the readme fragment header lines.
Can you please squash into a single commit (hint - if you `commit --amend` squashing won't be needed).

yanchirino commented:
> Thank you. The README header dos not translate properly to HTML. I believe the issue is the `\` leading character in the readme fragment header lines. Can you please squash into a single commit (hint - if you `commit --amend` squashing won't be needed).

@dreispt Thanks for the feedback! I’ve made the changes and I’m looking forward to your review.

mbcosta commented:
Hi @yanchirino  , I saw your email in the Contributors List, the main responsible for an OCA repository are the **PSCs** you can reach they by doing

cc @OCA/local-colombia-maintainers  (Normaly at the end of a message in PRs or Issue )

They are the people with rights / access / permission / credentials to make the Merge of your PR.  

About your PR there are some points:

* The python files should be have in first line a License Header, for example:

```bash
# Copyright 2025-TODAY Your_Company - Your_Name <your_email>
# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
```

Excepted init.py in the cases where has only imports.

The XML files can also have License Header, for example

```bash
<?xml version="1.0" encoding="utf-8" ?>
<!--
  Copyright (C) 2025-Today Your_Company - Your_Name <your_email>
  License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
-->
```

* There are an OCA repository for Currency related things https://github.com/OCA/currency/tree/18.0 and a specfic module to be use as base for exchange cases https://github.com/OCA/currency/tree/18.0/currency_rate_update

<img width="1171" height="381" alt="image" src="https://github.com/user-attachments/assets/f7659ee5-453c-4b60-b2f6-31470759be66" />

I don't know if its fit for your case, but you can analise in order to reduce code in your module by using it, if its fit in Brazilian Localization project has a example of how you can make https://github.com/OCA/l10n-brazil/tree/16.0/l10n_br_currency_rate_update

yanchirino commented:
> Hi @yanchirino , I saw your email in the Contributors List, the main responsible for an OCA repository are the **PSCs** you can reach they by doing
> 
> cc @OCA/local-colombia-maintainers (Normaly at the end of a message in PRs or Issue )
> 
> They are the people with rights / access / permission / credentials to make the Merge of your PR.
> 
> About your PR there are some points:
> 
> * The python files should be have in first line a License Header, for example:
> 
> ```shell
> # Copyright 2025-TODAY Your_Company - Your_Name <your_email>
> # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
> ```
> 
> Excepted init.py in the cases where has only imports.
> 
> The XML files can also have License Header, for example
> 
> ```shell
> <?xml version="1.0" encoding="utf-8" ?>
> <!--
>   Copyright (C) 2025-Today Your_Company - Your_Name <your_email>
>   License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
> -->
> ```
> 
> * There are an OCA repository for Currency related things https://github.com/OCA/currency/tree/18.0 and a specfic module to be use as base for exchange cases https://github.com/OCA/currency/tree/18.0/currency_rate_update
> 
> <img alt="image" width="1171" height="381" src="https://private-user-images.githubusercontent.com/6341149/497486630-f7659ee5-453c-4b60-b2f6-31470759be66.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTk2MTMyOTgsIm5iZiI6MTc1OTYxMjk5OCwicGF0aCI6Ii82MzQxMTQ5LzQ5NzQ4NjYzMC1mNzY1OWVlNS00NTNjLTRiNjAtYjJmNi0zMTQ3MDc1OWJlNjYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTAwNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEwMDRUMjEyMzE4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDQzZTAwMDk5YTkwOWZlYjlmYzRlMWY0YjZlNTliMDI2ZWE0NTVmMmU4ZmQ5ZWQzYzU2YmNhZGU4NmU0OWE3MiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.VIRkf5sslOt6Z4j1YtdEV6OfWivP_63jftvSB6D5Xd4">
> I don't know if its fit for your case, but you can analise in order to reduce code in your module by using it, if its fit in Brazilian Localization project has a example of how you can make https://github.com/OCA/l10n-brazil/tree/16.0/l10n_br_currency_rate_update

@mbcosta  Thanks for your comments and suggestions. I hadn’t considered using the base currency module, but it’s definitely a better option to stay aligned with the ecosystem. I’ve already made the required changes to adapt the provider.

cc @OCA/local-colombia-maintainers

yanchirino committed:
[ADD] l10n_co_currency_rate_update: Currency Rate from Superfinanciera

- Add automatic COP/USD exchange rate update from Superfinanciera
- Integrate with Superfinanciera SOAP web service

yanchirino commented:
Hi, what's the next step I should take to merge the code? Should I notify a specific person or a channel? Are more reviewers needed? It seems to me we're stuck, and I have more things to contribute.

CC
@OCA/local-colombia-maintainers
@dreispt 
@mbcosta

dreispt commented:
Maintainers are not responding, code looks good, I'm proceeding with the merge.
/oca bot merge nobump

yanchirino commented:
> Maintainers are not responding, code looks good, I'm proceeding with the merge. /oca bot merge nobump

@dreispt I’m really interested in contributing more actively to this project. Could you please let me know how I can become a maintainer, or get a role that allows me to merge code once it’s been reviewed and approved by others? Are there any requirements I should meet or events I need to participate in to be considered?

Thanks for your time and for all the effort!

etobella commented:
/ocabot merge nobump

etobella commented:
@yanchirino if you want to become a maintainer, you need to add yourself here: 

https://github.com/OCA/repo-maintainer-conf/blob/master/conf/psc/l10n.yml#L111

dreispt commented:
It looks my merge command didnt work.
Thank you Enric.

etobella commented:
There was a small typographic error. It can happen (once I wrote down ocabat)
